Sources/Core/Needs Conversion/ServerConfigurationService.swift

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+typealias ServerConfigurationResult = Result<ServerConfiguration, Error>
+typealias ServerConfigurationCompletion = (ServerConfigurationResult) -> Void
+
+protocol ServerConfigurationServicing {
+  var serverConfiguration: ServerConfiguration { get }
+}
+
+class ServerConfigurationService: ServerConfigurationServicing {
+  static var shared = ServerConfigurationService()
+
+  private let oneHourInSeconds = TimeInterval(60 * 60)
+  private let loadTimeout = TimeInterval(4.0)
+
+  private(set) var settings: SettingsManaging
+  private(set) var graphConnectionProvider: GraphConnectionProviding
+  private(set) var store: ServerConfigurationStore
+  private(set) var logger: Logging
+
+  private var appIdentifier: String {
+    return settings.appIdentifier ?? "Missing app identifier. Please add one in Settings."
+  }
+
+  var serverConfiguration: ServerConfiguration {
+    didSet {
+      store.cache(serverConfiguration)
+    }
+  }
+
+  /**
+   A computed default for ServerConfiguration. Attempts to use the current
+   application identifier from the settings but will default to a message
+   with instructions to add an identifier
+   */
+  var defaultServerConfiguration: ServerConfiguration {
+      return ServerConfiguration(appID: appIdentifier)
+  }
+
+  var isRequeryFinishedForAppStart: Bool = false
+  private var isLoading: Bool = false
+
+  private var dialogFlowsField: String {
+    let operatingSystemVersion = ProcessInfo.processInfo.operatingSystemVersion
+    return [
+      FieldKeys.dialogFlows,
+      ".os_version(",
+      "\(operatingSystemVersion.majorVersion)",
+      ".\(operatingSystemVersion.minorVersion)",
+      ".\(operatingSystemVersion.patchVersion)",
+      ")"
+    ].joined()
```

Comment:
It's true. I mainly did this for readability and to keep with the linter's preference for shorter lines. It may be more clear as a concatenated string though.

Reply:
Changing to use concatenated string.